pkg/translator/sumoicmarshaler/sumo_ic_marshaler.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumoicmarshaler
+
+import (
+	"encoding/json"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	SourceCategoryKey = "_sourceCategory"
+	SourceHostKey     = "_sourceHost"
+	SourceNameKey     = "log.file.path_resolved"
+)
+
+type SumoICMarshaler struct {
+	format string
+}
+
+type fieldSet struct {
+	attributes map[string]interface{}
+}
+
+type icRecord struct {
+	Data           string   `json:"data"`
+	SourceName     string   `json:"sourceName"`
+	SourceHost     string   `json:"sourceHost"`
+	SourceCategory string   `json:"sourceCategory"`
+	Fields         fieldSet `json:"fields"`
+	Message        string   `json:"message"`
+}
+
+func (fields fieldSet) MarshalJSON() ([]byte, error) {
+	return json.Marshal(fields.attributes)
+}
+
+func transformLogsToIc(ld plog.Logs) ([]icRecord, error) {
+	icRecords := []icRecord{}
+	rls := ld.ResourceLogs()
+
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		sourceCategoryVal := ""
+		sourceCategory, exists := rl.Resource().Attributes().Get(SourceCategoryKey)
+		if exists {
+			sourceCategoryVal = sourceCategory.AsString()
+		}
+		sourceHostVal := ""
+		sourceHost, exists := rl.Resource().Attributes().Get(SourceHostKey)
+		if exists {
+			sourceHostVal = sourceHost.AsString()
+		}
+		ills := rl.ScopeLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			logs := ils.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				lr := logs.At(k)
+				sourceNameVal := ""
+				sourceName, exists := lr.Attributes().Get(SourceNameKey)
+				if exists {
+					sourceNameVal = sourceName.AsString()
+				}
+				fields := fieldSet{attributes: make(map[string]interface{})}
+				attrs := lr.Attributes().AsRaw()
+				for ak, av := range attrs {
+					fields.attributes[ak] = av
+				}
+
```

Comment:
So this is not how we currently process logs when ingesting them through OTLP source. We should be doing following:

1) Fields should be taken from resource-level attributes
2) Body should be based on logrecord-level attributes + logrecord body (if body is a raw string, it should be stored under a separate key)

Here is more detail: https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/OpenTelemetryConcepts.md

Reply:
Actually, I don't understand 2 point, should body be concatenated with logrecord level attributes?

Log record from Installed Collector looks as follows (body is under message key):

`{"data": "2022-05-30 08:20:54.133382 +0000 UTC","sourceName":"/Users/pdelewski/Projects/logwriter/data.log","sourceHost":"pdelewski-mac","sourceCategory":"logfile","fields":{},"message":"2022-05-26 14:15:13.115274 +0200 CEST m=+9807.933100239 log entry9800"}`

Let's say that we have log record attributes, how then above log record should look like?